File: postgres_39/all_struct_with_default/SourceDocAuthor.go
package dto

type Source_doc_author struct {
	Source_document_id string  `json:"source_document_id" default:"source"`
	Author_id          string  `json:"author_id" default:"source"`
	Active_ind         *string `json:"active_ind" default:""`
	Author_ba_id       *string `json:"author_ba_id" default:""`
	Author_first_name  *string `json:"author_first_name" default:""`
	Author_initial     *string `json:"author_initial" default:""`
	Author_last_name   *string `json:"author_last_name" default:""`
	Author_seq_no      *int    `json:"author_seq_no" default:""`
	Author_type        *string `json:"author_type" default:""`
	Effective_date     *string `json:"effective_date" default:""`
	Expiry_date        *string `json:"expiry_date" default:""`
	Ppdm_guid          *string `json:"ppdm_guid" default:""`
	Remark             *string `json:"remark" default:""`
	Source             *string `json:"source" default:""`
	Row_changed_by     *string `json:"row_changed_by" default:""`
	Row_changed_date   *string `json:"row_changed_date" default:""`
	Row_created_by     *string `json:"row_created_by" default:""`
	Row_created_date   *string `json:"row_created_date" default:""`
	Row_effective_date *string `json:"row_effective_date" default:""`
	Row_expiry_date    *string `json:"row_expiry_date" default:""`
	Row_quality        *string `json:"row_quality" default:""`
}

File: create_tables/source_doc_author.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table source_doc_author';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute("""
    create table source_doc_author (
        SOURCE_DOCUMENT_ID VARCHAR(40) PRIMARY KEY NOT NULL,
        AUTHOR_ID VARCHAR(40) NOT NULL,
        ACTIVE_IND VARCHAR(1),
        AUTHOR_BA_ID VARCHAR(40),
        AUTHOR_FIRST_NAME VARCHAR(30),
        AUTHOR_INITIAL VARCHAR(30),
        AUTHOR_LAST_NAME VARCHAR(30),
        AUTHOR_SEQ_NO INT,
        AUTHOR_TYPE VARCHAR(40),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        PPDM_GUID VARCHAR(38),
        REMARK VARCHAR(2000),
        SOURCE VARCHAR(40),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")
